DXRMissions/DeusEx/Classes/DXRMissionsM02.uc

```diff
@@ -274,32 +275,36 @@ function AfterMoveGoalToLocation(Goal g, GoalLocation Loc)
     local #var(prefix)ComputerPersonal cp;
     local DXRPasswords passwords;
     local int i,rad,height;
-    local bool RevisionMaps;
-
-    RevisionMaps = class'DXRMapVariants'.static.IsRevisionMaps(player());
+    local AmbientSoundTriggered ast1,ast2;
 
     if (g.name=="Generator"){
-        if (!RevisionMaps){
-            if (Loc.Name=="Warehouse" || Loc.Name=="Sewer" || Loc.Name=="3rd Floor" || Loc.Name=="4th Floor"){ //These ones lie horizontal
-                rad=175;
-                height=100;
-            } else { //The rest stand upright
-                rad=120;
-                height=131;
-            }
-            class'DXRHoverHint'.static.Create(self, "NSF Generator", g.actors[0].a.Location, rad, height, g.actors[0].a);
-            a = Spawn(class'DynamicBlockMonsters',,, g.actors[0].a.Location);
-            a.SetBase(g.actors[0].a);
-            a.SetCollisionSize(rad, height);
-        } else {
-            class'DXRHoverHint'.static.Create(self, "NSF Generator", g.actors[0].a.Location, 175, 140, g.actors[0].a);
+        if (Loc.Name=="Warehouse" || Loc.Name=="Sewer" || Loc.Name=="3rd Floor" || Loc.Name=="4th Floor"){ //These ones lie horizontal
+            rad=175;
+            height=100;
+        } else { //The rest stand upright
+            rad=120;
+            height=131;
         }
+        class'DXRHoverHint'.static.Create(self, "NSF Generator", g.actors[0].a.Location, rad, height, g.actors[0].a);
+        a = Spawn(class'DynamicBlockMonsters',,, g.actors[0].a.Location);
+        a.SetBase(g.actors[0].a);
+        a.SetCollisionSize(rad, height);
 
         for(i=0; i<ArrayCount(g.actors); i++) {
             if( AmbientSoundTriggered(g.actors[i].a) != None) {
                 AmbientSoundTriggered(g.actors[i].a).SoundRadius = 1600;
             }
         }
+
+        //Find the original alarm sound
+        foreach AllActors(class'AmbientSoundTriggered',ast1,'WarningSound'){ break; }
+
+        //Spawn an alarm sound at the generator as well (the original one goes with the computer)
+        ast2 = Spawn(class'AmbientSoundTriggered',,'WarningSound', g.actors[0].a.Location);
+        ast2.savedSound=ast1.savedSound; //Revision didn't like referencing the sound by name, so just copy it from the original
```

Comment:
it probably isn't loaded until the level data is loaded? a dynamic load object probably would've worked, but this is better